supabase/migrations/24_billing_historicals.sql

```diff
@@ -0,0 +1,230 @@
+
+-- Always use a transaction, y'all.
+begin;
+
+-- Historical record of tenant billing statements. This structure
+-- comes from the return value of `billing_report_202308`.
+create table billing_historicals (
+    billed_prefix       catalog_tenant  not null,
```

Comment:
optional: should this be `tenant` ? I'm cool with `billed_prefix` instead, though, and am on the fence myself.

Reply:
So I actually had it as `tenant` originally, then changed it to `billed_prefix` to match the response coming back from `billing_report_202308()` so the UI could treat them interchangeably... but now I'm realizing that I could have just aliased the field when querying since postgrest lets you do that. I do think that `tenant` is slightly more correct -- we want to bill based on tenants, and we populate this table by iterating over tenants. I'm very slightly leaning towards changing it

Reply:
if the UI just fetches out the `report` field, that will contain `billed_prefix` in the same way that the RPC report does, if that's helpful for consolidation?

Reply:
Yeah that's a good point. I don't really love this duplication but at least we talked about it and decided that it's worth it to leave the exact shape of `report` up to the report generator instead of making it explicit here.